rest/test/db/CatapultDb_spec.js

```diff
@@ -1512,16 +1518,657 @@ describe('catapult db', () => {
+				{ _id: createObjectId(20), meta: { numTransactions: 1 }, block: { version: 3, type: 2 } }
+			]);
+			const transactions = () => ([
+				{ _id: createObjectId(30), meta: { height: 1 }, transaction: { type: 3 } }
+			]);
+
+			const runCollectionTest = (collectionName, objectId) => {
+				it(`respects collection: ${collectionName}`, () => {
+					const sortConditions = { $sort: { _id: 1 } };
+					const options = { pageSize: 10, pageNumber: 1 };
+
+					// Act + Assert:
+					return runDbTest(
+						{ accounts: accounts(), blocks: blocks(), transactions: transactions() },
+						db => db.queryPagedDocuments_2([], [], sortConditions, collectionName, options),
+						page => {
+							expect(page.data.length).to.equal(1);
+							expect(page.data[0].id).to.deep.equal(createObjectId(objectId));
+						}
+					);
+				});
+			};
+
+			[
+				{ collectionName: 'accounts', id: 10 },
+				{ collectionName: 'blocks', id: 20 },
+				{ collectionName: 'transactions', id: 30 }
+			].forEach(testCase => {
+				runCollectionTest(testCase.collectionName, testCase.id);
+			});
+		});
+
+		describe('options', () => {
+			const blocks = numBlocks => {
+				const resultBlocks = [];
+				for (let i = 0; numBlocks > i; i++)
+					resultBlocks.push({ _id: createObjectId(i), meta: { hash: 0x0100 }, block: { type: 1 } });
+
+				return resultBlocks;
+			};
+
+			describe('respects page size', () => {
+				it('page size: 25', () => {
+					const options = { pageSize: 25, pageNumber: 1 };
+
+					// Act + Assert:
+					return runDbTest(
+						{ blocks: blocks(25 + 10) },
+						db => db.queryPagedDocuments_2([], [], { $sort: { _id: 1 } }, 'blocks', options),
+						page => {
+							expect(page.data.length).to.equal(25);
+							page.data.forEach((element, i) => {
+								expect(element.id).to.deep.equal(createObjectId(i));
+							});
+							expect(page.paging).to.deep.equal({
+								totalEntries: 25 + 10, pageNumber: 1, pageSize: 25
+							});
+						}
+					);
+				});
+			});
+
+			describe('respects page number', () => {
+				const runPageNumberTest = (pageNumber, expectedNumberOfElements) => {
+					it(`page number: ${pageNumber}`, () => {
+						const options = { pageSize: 10, pageNumber };
+
+						// Act + Assert:
+						return runDbTest(
+							{ blocks: blocks(32) },
+							db => db.queryPagedDocuments_2([], [], { $sort: { _id: 1 } }, 'blocks', options),
+							page => {
+								expect(page.data.length).to.equal(expectedNumberOfElements);
+								page.data.forEach((element, i) => {
+									expect(element.id).to.deep.equal(createObjectId(((pageNumber - 1) * 10) + i));
+								});
+								expect(page.paging).to.deep.equal({
+									totalEntries: 32, pageNumber, pageSize: 10
+								});
+							}
+						);
+					});
+				};
+
+				runPageNumberTest(1, 10);
+				runPageNumberTest(2, 10);
+				runPageNumberTest(3, 10);
+				runPageNumberTest(4, 2);
+				runPageNumberTest(5, 0);
+			});
+		});
+	});
+
+	describe('transactions', () => {
```

Comment:
missing test without any filters (i think? unsure if i've commented this elsewhere)